src/services/db/GitFileSystemService.ts

```diff
@@ -112,6 +115,39 @@ export default class GitFileSystemService {
       })
   }
 
+  // Ensure that the repository is in the BRANCH_REF branch
+  ensureCorrectBranch(repoName: string): ResultAsync<true, GitFileSystemError> {
```

Comment:
this function's job to ensure a repo is on a correct branch. shouldn't it take in the 2 dependencies - repoName and branchName as 2 arguments and allow upstream to call `ensureCorrectBranch(repoName, branchName)`

Reply:
Hmm for this one, we are making sure that the branch is at `staging`, as the CMS does not work on any other branch when performing changes. Hence, it only takes in one argument in this case.